ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -795,17 +795,28 @@ abstract class EventStorageBackendTemplate(
   }
 
   override def maxEventSequentialIdOfAnObservableEvent(
-      offset: Offset
-  )(connection: Connection): Option[Long] = {
+      offsetInclusive: Offset
+  )(connection: Connection): Long = {
+    val ledgerEnd = ledgerEndCache()
     import com.daml.platform.store.backend.Conversions.OffsetToStatement
     SQL"""
-         SELECT
-            event_sequential_id_last
-         FROM
-            participant_transaction_meta
-         WHERE
-            event_offset = (SELECT MAX(event_offset) FROM participant_transaction_meta WHERE event_offset <= $offset)
-       """.as(get[Long](1).singleOpt)(connection)
+     SELECT
+        event_sequential_id_first
+     FROM
+        participant_transaction_meta
+     WHERE
+        event_offset > $offsetInclusive
+        AND event_offset <= ${ledgerEnd._1}
```

Comment:
it is correctness. in the query before we did not need to constrain to the ledger end since the query was looking backwards from an offset which is for sure smaller or equal than the ledger end. in this case we are looking ahead, meaning: doing this unbounded we would look into the parallel ingestion zone, where it might happen we see a meta entry which is already ingested, but other meta entries might follow, which are earlier, and picking up that would be just wrong.